kasa/iot/modules/motion.py

```diff
@@ -20,6 +22,67 @@ class Range(Enum):
     Near = 2
     Custom = 3
 
+    def __str__(self) -> str:
+        return self.name
+
+
+@dataclass
+class PIRConfig:
+    """Dataclass representing a PIR sensor configuration."""
+
+    enabled: bool
+    adc_min: int
+    adc_max: int
+    range: Range
+    threshold: int
+
+    @property
+    def adc_mid(self) -> int:
+        """Compute the ADC midpoint from the configured ADC Max and Min values."""
+        return math.floor(abs(self.adc_max - self.adc_min) / 2)
+
+
+@dataclass
+class PIRStatus:
```

Comment:
I think this should take `PIRConfig` in it's constructor as `init_only` and all the `get_` methods can then attributes. If you include the mashumaro `DataClassMixin` the data class will then get `to_dict()` method and you can set all the attributes in post init. I think that is more intuitive than returning a status that then needs to be provided config to get more values.